x-pack/plugins/ml/public/application/components/data_grid/use_column_chart.tsx

```diff
@@ -0,0 +1,442 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+import { BehaviorSubject } from 'rxjs';
+import React from 'react';
+
+import { useObservable } from 'react-use';
+
+import { euiPaletteColorBlind, EuiDataGridColumn } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { KBN_FIELD_TYPES } from '../../../../../../../src/plugins/data/public';
+
+import { stringHash } from '../../../../common/util/string_utils';
+
+import { NON_AGGREGATABLE } from './common';
+
+export const hoveredRow$ = new BehaviorSubject<any | null>(null);
+
+const BAR_COLOR = euiPaletteColorBlind()[0];
+const BAR_COLOR_BLUR = euiPaletteColorBlind({ rotations: 2 })[10];
+const MAX_CHART_COLUMNS = 20;
+
+type XScaleType = 'ordinal' | 'time' | 'linear' | undefined;
+const getXScaleType = (kbnFieldType: KBN_FIELD_TYPES | undefined): XScaleType => {
+  switch (kbnFieldType) {
+    case KBN_FIELD_TYPES.BOOLEAN:
+    case KBN_FIELD_TYPES.IP:
+    case KBN_FIELD_TYPES.STRING:
+      return 'ordinal';
+    case KBN_FIELD_TYPES.DATE:
+      return 'time';
+    case KBN_FIELD_TYPES.NUMBER:
+      return 'linear';
+  }
+};
+
+const getFieldType = (schema: EuiDataGridColumn['schema']): KBN_FIELD_TYPES | undefined => {
+  if (schema === NON_AGGREGATABLE) {
+    return undefined;
+  }
+
+  let fieldType: KBN_FIELD_TYPES;
+
+  switch (schema) {
+    case 'datetime':
+      fieldType = KBN_FIELD_TYPES.DATE;
+      break;
+    case 'numeric':
+      fieldType = KBN_FIELD_TYPES.NUMBER;
+      break;
+    case 'boolean':
+      fieldType = KBN_FIELD_TYPES.BOOLEAN;
+      break;
+    case 'json':
+      fieldType = KBN_FIELD_TYPES.OBJECT;
+      break;
+    default:
+      fieldType = KBN_FIELD_TYPES.STRING;
+  }
+
+  return fieldType;
+};
+
+interface NumericColumnStats {
+  interval: number;
+  min: number;
+  max: number;
+}
+type NumericColumnStatsMap = Record<string, NumericColumnStats>;
+const getAggIntervals = async (
+  indexPatternTitle: string,
+  api: any,
+  query: any,
+  columnTypes: EuiDataGridColumn[]
+): Promise<NumericColumnStatsMap> => {
+  const numericColumns = columnTypes.filter((cT) => {
+    const fieldType = getFieldType(cT.schema);
+    return fieldType === KBN_FIELD_TYPES.NUMBER || fieldType === KBN_FIELD_TYPES.DATE;
+  });
+
+  if (numericColumns.length === 0) {
+    return {};
+  }
+
+  const minMaxAggs = numericColumns.reduce((aggs, c) => {
+    const id = stringHash(c.id);
+    aggs[id] = {
+      stats: {
+        field: c.id,
+      },
+    };
+    return aggs;
+  }, {} as Record<string, object>);
+
+  let respStats: any;
+  try {
+    respStats = await api.esSearch({
+      index: indexPatternTitle,
+      size: 0,
+      body: {
+        query,
+        aggs: minMaxAggs,
+        size: 0,
+      },
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
```

Comment:
This catch statement seems redundant.

Reply:
Removed the whole try/catch block, errors will get caught by an outer try/catch anyway 82447ae.